openedx/core/djangoapps/programs/tests/test_tasks.py

```diff
@@ -0,0 +1,244 @@
+        mock_get_user_credentials.return_value = [
+            self.make_credential_result(status='awarded', credential={'program_id': 1}),
+            self.make_credential_result(status='awarded', credential={'course_id': 2}),
+            self.make_credential_result(status='revoked', credential={'program_id': 3}),
+        ]
+
+        result = tasks.get_awarded_certificate_programs(student)
+        self.assertEqual(mock_get_user_credentials.call_args[0], (student, ))
+        self.assertEqual(result, [1])
+
+
+class AwardProgramCertificateTestCase(TestCase):
+    """
+    Test the award_program_certificate function
+    """
+
+    @httpretty.activate
+    def test_award_program_certificate(self):
+        """
+        Ensure the correct API call gets made
+        """
+        student = UserFactory(username='test-username')
+        test_client = EdxRestApiClient('http://test-server', jwt='test-token')
+        httpretty.register_uri(
+            httpretty.POST,
+            'http://test-server/user_credentials/',
+        )
+        tasks.award_program_certificate(test_client, student, 123)
+        self.assertEqual(httpretty.last_request().body, json.dumps({'program_id': 123, 'username': student.username}))
+
+
+@ddt.ddt
+@mock.patch('openedx.core.djangoapps.programs.tasks.award_program_certificate')
+@mock.patch('openedx.core.djangoapps.programs.tasks.get_awarded_certificate_programs')
+@mock.patch('openedx.core.djangoapps.programs.tasks.get_completed_programs')
+@mock.patch('openedx.core.djangoapps.programs.tasks.get_completed_courses')
+class AwardProgramCertificatesTestCase(TestCase, ProgramsApiConfigMixin, CredentialsApiConfigMixin):
+    """
+    Tests for the 'award_program_certificates' celery task.
+    """
+
+    def setUp(self):
+        super(AwardProgramCertificatesTestCase, self).setUp()
+        self.programs_config = self.create_programs_config()
+        self.credentials_config = self.create_credentials_config()
+        self.student = UserFactory.create(username='test-student')
+
+        ClientFactory.create(name='programs')
+        ClientFactory.create(name='credentials')
+        UserFactory.create(username='FIXME-NO-SERVICE-USER')  # temporary hack for credentials API call
+
+    def test_completion_check(
+            self,
+            mock_get_completed_courses,
+            mock_get_completed_programs,
+            mock_get_awarded_certificate_programs,  # pylint: disable=unused-argument
+            mock_award_program_certificate,  # pylint: disable=unused-argument
+    ):
+        """
+        Checks that the Programs API is used correctly to determine completed
+        programs.
+        """
+        completed_courses = [
+            {'course_id': 'course-1', 'type': 'verified'},
+            {'course_id': 'course-2', 'type': 'prof-ed'},
+        ]
+        mock_get_completed_courses.return_value = completed_courses
+
+        tasks.award_program_certificates(self.student.username)
```

Comment:
The main concern of this test is that the call we make to programs contains the correct payload, specifically the result from `get_completed_courses`.  The return values from the other mocked helpers are not interesting, but I am validating that based on a certain set of completed courses, we will ask Programs the right question about completion.